src/utils/loadable.spec.ts

```diff
@@ -0,0 +1,50 @@
+import { setUpLoadableComponent, setComponentLoaded, componentLoaded } from "./loadable";
+
+describe("loadable", () => {
+  it("lifecycle", async () => {
+    const status = {
+      willLoad: false,
+      loaded: false,
+      promise: null
+    };
+    const fakeComponent: any = {
```

Comment:
I really like how this test is showing how the modules work together. If not too complex, we could do the same for other spec tests.

From that perspective, I think we could simplify the test and have it focus on its use case:

```ts
const afterDidLoad = fakeComponent.resolvesAfterDidLoad();

// assert afterDidLoad not resolved

fakeComponent.componentWillLoad();

// assert afterDidLoad not resolved

fakeComponent.componentDidLoad();

// assert afterDidLoad resolved
```

Alternatively, the individual utils could be tested as well:

```ts
// assert componentLoaded(fakeComponent) not resolved

setUpLoadableComponent(fakeComponent);
// assert componentLoaded(fakeComponent) not resolved

componentLoaded(fakeComponent);
// assert componentLoaded(fakeComponent) resolved
```

Reply:
> I really like how this test is showing how the modules work together. If not too complex, we could do the same for other spec tests.

Can we get a followup issue for this? 👍 

> // assert afterDidLoad not resolved

How can I assert the promise is not resolved? There's not a property on the promise. Is there some util for this?

Reply:
Here's one, _untested_, idea:

```ts
const afterLoad = jest.fn();
const load = componentLoaded(fakeComponent).then(afterLoad);

await waitForAnimationFrame();
assert(afterLoad).not.toHaveBeenCalled();

setUpLoadableComponent(fakeComponent);
await waitForAnimationFrame();
assert(afterLoad).not.toHaveBeenCalled();

componentLoaded(fakeComponent);
await waitForAnimationFrame();
assert(afterLoad).toHaveBeenCalled();
```

Reply:
done!